framework/build.mk

```diff
@@ -57,6 +57,16 @@ ifneq (,$(findstring mpi,$(cxx_compiler)))
 	cxx_compiler = $(shell $(libmesh_CXX) -show)
 endif
 
+# Check that we have Asio installed.
+ASIO_FILE := $(MOOSE_DIR)/framework/contrib/asio/include/asio.hpp
+ifneq ("$(wildcard $(ASIO_FILE))","")
+# If it is installed, make sure this is a 
+# build using C++11, if so, then turn on Asio Networking
+ifeq (,$(findstring "-std=gnu++11",$(libmesh_CXXFLAGS)))
```

Comment:
That value is determined by several configure tests (m4 scripts). Basically
we do it the old tried and true way. We compile several small test problems
and see if they work or not. What we find is that most compilers support a
subset of C++11 features unless you explicitly tell the compiler to obey
ANSI standards. It's tricky to get this behavior right so you don't want to
reinvent the wheel here.

On Thu, Dec 10, 2015 at 7:40 AM Alex McCaskey <notifications@github.com>
wrote:

> In framework/build.mk
> <https://github.com/idaholab/moose/pull/6037#discussion_r47233115>:
>
> > @@ -57,6 +57,16 @@ ifneq (,$(findstring mpi,$(cxx_compiler)))
> >  	cxx_compiler = $(shell $(libmesh_CXX) -show)
> >  endif
> >
> > +# Check that we have Asio installed.
> > +ASIO_FILE := $(MOOSE_DIR)/framework/contrib/asio/include/asio.hpp
> > +ifneq ("$(wildcard $(ASIO_FILE))","")
> > +# If it is installed, make sure this is a
> > +# build using C++11, if so, then turn on Asio Networking
> > +ifeq (,$(findstring "-std=gnu++11",$(libmesh_CXXFLAGS)))
>
> @permcody <https://github.com/permcody> @jwpeterson
> <https://github.com/jwpeterson> So you guys are saying that checking for
> -std=gnu+11 won't work for some of the other compilers? How do you check in
> the Makefile for the LIBMESH_HAVE_CXX11 constant?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/idaholab/moose/pull/6037/files#r47233115>.
>